TFP-6071: Overstyring og omskriving fakta fødsel

remhol opened the pull request:
* Ny tjeneste for fakta fødsel
* Ny rest tjeneste for fakta fødsel med nytt get-endepunkt
* Ny dto for fremvisning
* Overstyring-dto og nytt aksjonspunkt
* Håndterer for overstyring

remhol committed:
TFP-6071: Overstyring og omskriving fakta fødsel
* Ny tjeneste for fakta fødsel
* Ny rest tjeneste for fakta fødsel med nytt get-endepunkt
* Ny dto for fremvisning
* Overstyring-dto og nytt aksjonspunkt
* Håndterer for overstyring

remhol committed:
TFP-6071: Refaktorering

remhol committed:
TFP-6071: Refaktorering

remhol committed:
TFP-6071: Fikser test

remhol committed:
Merge branch 'master' into TFP-6071

remhol committed:
TFP-6071: Kan ikke overstyre hvis kilde freg

remhol committed:
TFP-6071: Feil bruk av kilde

remhol committed:
TFP-6071: Oppdatert likeBarn-sjekken til å håndtere ulik rekkefølge

jolarsen commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/fødsel/dto/Kilde.java:
```diff
@@ -0,0 +1,7 @@
+package no.nav.foreldrepenger.web.app.tjenester.behandling.fødsel.dto;
+
+public enum Kilde {
```
Skrev gjerne helt ut :-) SAKSBEHANDLER er mer lesbart enn SBH og FOLKEREGISTER enn FREG

jolarsen commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/RestImplementationClasses.java:
```diff
@@ -113,6 +114,7 @@ public static Collection<Class<?>> getImplementationClasses() {
         classes.add(LosRestTjeneste.class);
         classes.add(RedirectToRegisterRestTjeneste.class);
         classes.add(OppgaverRestTjeneste.class);
+        classes.add(FødselRestTjeneste.class);
```
Når dere legger til denne, så dukker det lettere opp spørsmål om det kommer en ekstra tjeneste for Adopsjon og Omsorgsovertakelse?
Regner med målet blir er å rydde opp i FamiliehendelseRestTjeneste og tilhørende Dtos - som er en liten hodepine. Sammen med SoknadDto som man bør se på hvordan brukes frontend.
Først var det SoknadDto + RegisterDto +++, så ble søknad delt i domene/grunnlag og det kom GrunnlagDtos. SoknadDto er siste rest
FamilieHendelseDto og SoknadDto bruker begge JsonSubTypes

jolarsen commented on behandlingslager/domene/src/main/java/no/nav/foreldrepenger/behandlingslager/behandling/aksjonspunkt/AksjonspunktKodeDefinisjon.java:
```diff
@@ -58,6 +58,7 @@ public class AksjonspunktKodeDefinisjon {
     public static final String OVERSTYRING_AV_BEREGNINGSAKTIVITETER_KODE = "6014";
     public static final String OVERSTYRING_AV_BEREGNINGSGRUNNLAG_KODE = "6015";
     public static final String OVERSTYRING_AV_DEKNINGSGRAD_KODE = "6016";
+    public static final String OVERSTYRING_AV_FAKTA_OM_FØDSEL_KODE = "6019";
```
Sikre på at dere vil ha en overstyring for alle datoer, vs å skille overstyring av termindato og overstyr fødselsdato? Det er ganske ulike use-cases - termindato ifm permatur og føsdselsdato ifm utland.
Se ellers ForvaltningSøknadRestTjeneste.

jolarsen commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/fødsel/FaktaFødselTjeneste.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    public void overstyrFaktaOmFødsel(BehandlingReferanse ref, OverstyringFaktaOmFødselDto dto) {
+        // TODO: Legg til håndtering av om verdien kan endres
```
Generelt - dersom det finnes fødsels/dødsdato fra FREG så skal de ikke kunne overstyres. Dvs dersom bekreftetVersjon finnes - ikke overstyring av fødselsdatoer. Hvis det er avvik så skal man melde fra til Folkeregisteret på eget skjema eller be bruker kontakte FREG.
Overstyring av fødsels/dødsdato er aktuelt
- Det finnes overstyrt fødselsdato (etter en 5027 sjekk manglende fødsel)
- Man ikke ønsker å vente til 5027 oppstår pga etterkontroll - finnes bare søknadversjon

Når det gjelder overstyring av termindato
- dersom det finnes overstyrtHendelse - helt OK
- dersom det finnes bekreftetVersjon med barn - dvs FREG - litt usikker på om vi ønsker en overstyrt hendelse vs en "patch"-operasjon lik /endreTermindato - men bare termin-delen

Skal tenke mer på å skille ut registerdata som en helt egen ting - ikke en FamHendelseEnt.

jolarsen commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/fødsel/FaktaFødselTjeneste.java:
```diff
@@ -0,0 +1,111 @@
+
+        // TODO: Case: Født i utlandet, ikke registrert i FREG. Barn dør like etter fødsel. Dødsdato må overstyres
+
+        // TODO: Undersøk prematur dødfødsel, hva skjer da med freg og denne overstyringen?
```
Det håndteres i FREG. Hvis det ikke finnes i freg er det enten utland eller så tidlig at man ikke har rett på ytelse.

remhol committed:
TFP-6071: Fjernet todos etter avklaringer og skrevet ut kilder

thao1 committed:
Merge branch 'master' into TFP-6071

thao1 committed:
Merge branch 'master' into TFP-6071

thao1 committed:
TFP-6071: Lagt til enhetstest for hentFaktaFødsel.
- Endret fra en liste med AvklartBarnDto til bare et barn i FaktaFødselDto barn.

sirimykland committed:
Chore/tfp 6071 legg til utstedt dato i gjeldende (#7375)

legg til utstedtdato i gjeldende

thao1 committed:
TFP-6071: Lagt til logikk for hentFaktaOmFødsel og tilhørende enhetstest.

sirimykland committed:
Merge branch 'master' into TFP-6071

thao1 committed:
TFP-6071: Lagt til antallBarn for søknad og gjeldende i fødselDto.

thao1 committed:
TFP-6071: Oppdaterer enhetstester med antall barn.

thao1 committed:
TFP-6071: Fiks på mapTermindato og mapUtstedtdato når disse er null.

thao1 committed:
TFP-6071: Lager vår egen BarnHendelseData og benytter det istedet.
- Fjernet koder for overstyring som ikke skal være med.

thao1 committed:
TFP-6071: Fiks på at gjeldendeBarn liste skal inneholde riktig antall barn og kilde.

thao1 committed:
Merge branch 'master' into TFP-6071

palfi commented on web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/fødsel/FødselRestTjeneste.java:
```diff
@@ -0,0 +1,60 @@
+
+@Path(FødselRestTjeneste.BASE_PATH)
+@ApplicationScoped
+@Produces(MediaType.APPLICATION_JSON)
```
Trenger nok ikke å ha produces både på klassen og på metoden

palfi commented:
Må tenke over hva som skjer hvis behandlingen ikke har noe familiehendelse og denne tjenesten kalles. Hvordan ser fakta om fødsel ut før man har punchet papirsøknad? Hvordan toggler vi at panelet vises eller ikke

thao1 committed:
TFP-6071: Fiks etter QA.
- Refaktorerer FaktaFødselTjeneste koder slik at det blir mer lesbart.

thao1 commented:
> Kanskje fjerne overstyringsendringene fra første merge, bare ny restttjeneste. Så kan vi ta overstyringen neste?

Det har jeg allerede gjort! Eller har du sett noe sted at de ikke er fjernet? 
Ahh, ser at OverstyringFaktaOmFødselDto ikke er fjernet.

thao1 committed:
TFP-6071: Fjernet OverstyringFaktaOmFødselDto da det ikke er i bruk ennå.

thao1 committed:
Merge branch 'master' into TFP-6071

thao1 commented:
t

thao1 committed:
TFP-6071: legg til lenke for fødselsfakta når familiehendelse finnes.

Legger til en REST-lenke for fakta om fødsel i DTO'en kun når det finnes familiehendelse for behandlingen. Dette sikrer at frontend kun får tilgang til fødselsdata når det er relevant.

thao1 committed:
TFP-6071: Fjernet OVERSTYRING_AV_FAKTA_OM_FØDSEL_KODE fra AksjonspunktKodeDefinisjon.
